internal/services/network/subnet_resource_test.go

```diff
@@ -126,6 +126,24 @@ func TestAccSubnet_disappears(t *testing.T) {
 	})
 }
 
+func TestAccSubnet_defaultOutbound(t *testing.T) {
+	dataInternal := acceptance.BuildTestData(t, "azurerm_subnet", "internal")
+	dataPublic := acceptance.BuildTestData(t, "azurerm_subnet", "public")
+	r := SubnetResource{}
+
+	dataInternal.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.defaultOutbound(dataInternal),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(dataInternal.ResourceName).ExistsInAzure(r),
+				check.That(dataInternal.ResourceName).Key("default_outbound_access_enabled").HasValue("false"),
+				check.That(dataPublic.ResourceName).ExistsInAzure(r),
+				check.That(dataPublic.ResourceName).Key("default_outbound_access_enabled").HasValue("true"),
+			),
+		},
+	})
```

Comment:
we're only testing the Create flow here, but we support Update above, so can we update this to provision a Private subnet, change it to Public and go back again?

```suggestion
	data := acceptance.BuildTestData(t, "azurerm_subnet", "test")
	r := SubnetResource{}

	dataInternal.ResourceTest(t, r, []acceptance.TestStep{
		{
			Config: r.defaultOutbound(data, false),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("default_outbound_access_enabled").HasValue("false"),
			),
		},
		{
			Config: r.defaultOutbound(data, true),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("default_outbound_access_enabled").HasValue("true"),
			),
		},
		{
			Config: r.defaultOutbound(data, false),
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
				check.That(data.ResourceName).Key("default_outbound_access_enabled").HasValue("false"),
			),
		},
		})
```

Reply:
As mentioned above, there is no update so I removed the update code.